Src/BootCamp.Chapter/ToggleableGrid2d.cs

```diff
@@ -1,16 +1,47 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.ComTypes;
+
+namespace BootCamp.Chapter
 {
     public class ToggleableGrid2D : IToggleableGrid
     {
         private readonly IGridClearer _gridClearer;
+        private bool[,] _boolArray;
 
-        public ToggleableGrid2D(bool[,] toggles, IGridClearer gridClearer)
+        public ToggleableGrid2D(bool[,] boolArray
+            , IGridClearer gridClearer)
         {
             _gridClearer = gridClearer;
+            _boolArray = boolArray;
         }
 
         public void Toggle(int x, int y)
         {
+            int expectedMinimumArrayLength = (x + 1) * (y + 1);
+
+            if (_boolArray.Length < expectedMinimumArrayLength) throw new IndexOutOfRangeException("Index was outside of the bounds.");
+
+            _boolArray[x, y] = !_boolArray[x, y];
+            PrintArray();
+        }
+
+        private void PrintArray()
+        {
+            _gridClearer.Clear();
+
+            for (int i = 0; i < _boolArray.GetLength(0); i++)
+            {
+                for (int j = 0; j < _boolArray.GetLength(1); j++)
+                {
+                    Console.Write(_boolArray[i, j] == true ? "■" : " ");
```

Comment:
these magic strings should go as constants all the way at the top of the class. It's likely to change, but you don't want to look for every place where it was used.